lib/absinthe/schema.ex

```diff
@@ -145,17 +145,76 @@ defmodule Absinthe.Schema do
   alias Absinthe.Language
   alias __MODULE__
 
+  @doc """
+  Return the default middleware set for a field if none exists
+  """
+  def ensure_middleware([], %{identifier: identifier}, _) do
+    [{Absinthe.Middleware.MapGet, identifier}]
+  end
+  def ensure_middleware(middleware, _field, _object) do
+    middleware
+  end
+
   defmacro __using__(opts \\ []) do
     quote do
       use Absinthe.Schema.Notation, unquote(opts)
       import unquote(__MODULE__), only: :macros
+
       import_types Absinthe.Type.BuiltIns
+
       @after_compile unquote(__MODULE__)
       @behaviour unquote(__MODULE__)
+
+      @doc false
+      def __absinthe_middleware__(middleware, field, %{name: "__" <> _} = object) do
+        # if we have the double underscore prefix we're dealing with introspection
+        # types, which should use the built in default middleware
+        middleware
+        |> Absinthe.Schema.ensure_middleware(field, object)
+        |> __do_absinthe_middleware__(field, object)
+      end
+      def __absinthe_middleware__(middleware, field, object) do
+        __do_absinthe_middleware__(middleware, field, object)
+      end
+
+      defp __do_absinthe_middleware__(middleware, field, object) do
+        middleware
+        |> __MODULE__.middleware(field, object) # run field against user supplied function
+        |> Absinthe.Schema.ensure_middleware(field, object) # if they forgot to add middleware set the default
+      end
+
+      @doc false
+      def middleware(middleware, _field, _object) do
+        middleware
+      end
+
+      @doc false
+      def __absinthe_lookup__(key) do
+        key
+        |> __absinthe_type__
+        |> case do
+          %Absinthe.Type.Object{} = object ->
+            fields = Map.new(object.fields, fn
+              {identifier, field} ->
+                {identifier, %{field | middleware: __absinthe_middleware__(field.middleware, field, object)}}
+            end)
+
+            %{object | fields: fields}
+          type ->
+            type
+        end
+      end
+
+      @doc false
+      def plugins do
```

Comment:
I thought we agreed that a "plugin" was now exactly this -- middleware that requires this type of wiring-in?